forge-gui/src/main/java/forge/gamemodes/match/input/InputSelectTargets.java

```diff
@@ -360,8 +360,7 @@ private void addTarget(final GameEntity ge) {
         else if (ge instanceof Player) {
             getController().getGui().setHighlighted(PlayerView.get((Player) ge), true);
         }
-        final Integer val = targetDepth.get(ge);
-        targetDepth.put(ge, val == null ? Integer.valueOf(1) : Integer.valueOf(val.intValue() + 1) );
+        targets.add(ge);
 
         if (hasAllTargets()) {
```

Comment:
can the hasAllTargets() check popup a dialog to ask human player if the target is final, if not remove the selected target otherwise select the target and finish...

Reply:
om i have no idea to how code that right now? you are free to extend this MR if you want

Reply:
It can be added in a later MR since the additional confirmation is only needed if the sa can be paid for free (since you can cancel if theres inputpayment)